agent/job_runner.go

```diff
@@ -299,6 +300,34 @@ func NewJobRunner(l logger.Logger, scope *metrics.Scope, ag *api.AgentRegisterRe
 	return runner, nil
 }
 
+type hookExit struct {
+	ExitStatus   int
+	SignalReason string
+	Ok           bool
+}
+
+func (r *JobRunner) preExecHook(ctx context.Context, hookName string) hookExit {
+	exit := hookExit{Ok: true}
+	if hookPath, _ := hook.Find(r.conf.AgentConfiguration.HooksPath, hookName); hookPath != "" {
```

Comment:
[Totally optional suggestion] Now that preExecHook is its own function, early-return beckons...

```go
hookPath, _ := hook.Find(...)
if hookPath == "" {
    return exit
}
// Once we have ....
```

(The counterpoint is https://google.github.io/styleguide/go/decisions#indent-error-flow)

Reply:
yeah, i like this better than the alternative